### PR TITLE
Secure and simplify the s3 module

### PR DESCRIPTION
**Pull request by iuriaranda**
- Enable server-side encryption by default
- Restrict any public access to the bucket
- Remove variables to make the creation of s3 bucket and dynamodb table optional (not needed anymore)
- Remove variable to share the s3 bucket with other accounts (not needed anymore)

### CODE REVIEW

**Review by venturel — APPROVED** [review]

**Review by duboisph — APPROVED** [review]

**Comment by ringods** [comment]
@iuriaranda I've been away from AWS for a while. Can you elaborate why point 3 and 4 are not needed anymore?

BTW, can you still rewrite the README? The examples still refer to the situation where these removed variables existed.

**Comment by iuriaranda** [comment]
@ringods we've switched the approach for running Terraform on multiple AWS accounts (one per workspace).

If you recall, with the old approach, we created a single S3 bucket to hold the Terraform state files in an AWS account and shared it using the bucket policy with the rest of the accounts where Terraform would run. Then we created a DynamoDB lock table on each of the accounts, as it could not be shared like the S3 bucket.

We found this approach to be too complex and error-prone, and decided to switch to the Hashicorp recommended approach for [multi-account AWS architectures](https://www.terraform.io/docs/backends/types/s3.html#multi-account-aws-architecture), as we find it simpler and easier to maintain.

Basically we've decoupled the authentication for the S3 remote state from the one of the AWS provider.

I'll update the readme, thanks for the heads-up.
